src/components/DeleteConnection.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) [2020] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from 'react';
+import cockpit from 'cockpit';
+import ModalConfirm from './ModalConfirm';
+import { Button } from '@patternfly/react-core';
+
+const _ = cockpit.gettext;
+
+const DeleteConnection = ({ connection, deleteConnection }) => {
+    const [isConfirmationOpen, setConfirmationOpen] = useState(false);
+
+    const onClose = () => setConfirmationOpen(false);
+
+    const removeConnectionConfig = () => {
+        deleteConnection(connection);
+        onClose();
+    };
+
+    return (
+        <>
+            <Button variant="danger" onClick={() => setConfirmationOpen(true)}>{_("Delete")}</Button>
```

Comment:
Please, use an icon. Alternatively, you could use 'Remove' instead, but we need to avoid that ugly "DELETE" text :)

Reply:
I used 'Delete' because it looks the most common approach in other modules.  I added a TrashIcon instead of 'Delete' but @dgdavid was also not sure about being the best option there. It is used int podman for example.

Reply:
That's true. Since we are not really deleting / removing the interface, the _trash_ icon does not seems the best option in my view.

What about using the _undo_ icon with a [tooltip](https://www.patternfly.org/v4/components/tooltip) using the _Remove configuration_ or _Reset configuration_ instead? What do you think guys?

Reply:
I was already evaluating the use of a tooltip, and looks reasonable to me, but not sure about the `undo` icon, in fact, why not just use the button text `Remove configuration`? :stuck_out_tongue_winking_eye:

Reply:
Because, in my opinion, is _too much_. Seems that we are _calling to remove action_. Not sure if it is clear what I mean.

Reply:
> but not sure about the `undo` icon

Maybe better if we choose `fa-minus-circle` instead?